File: Data Structures & Algorithms/two-integer-sum/submission-0.rs
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut memory_map: HashMap<i32, i32> = HashMap::new();
        for (i, &num) in nums.iter().enumerate() {
            memory_map.insert(num, i as i32);
        }
        for (i, &num) in nums.iter().enumerate() {
            if let Some(&j) = memory_map.get(&(target - num)){
                if i as i32 != j{
                    return vec![i as i32, j];
                }
            }
        }
        vec![]
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        ret = 0
        for num in numSet:
            if not num-1 in numSet:
                length = 1
                while num+length in numSet:
                    length += 1

                ret = max(ret, length)

        return ret

                